pkg/p2p/libp2p/connections_test.go

```diff
@@ -40,8 +40,9 @@ func TestConnectDisconnect(t *testing.T) {
 	defer cancel()
 
 	s1, overlay1 := newService(t, 1, libp2pServiceOpts{})
-
 	s2, overlay2 := newService(t, 1, libp2pServiceOpts{})
+	defer s1.Close()
```

Comment:
All defers that are added in this PR are not needed as they are done in newService' t.Cleanup call https://github.com/ethersphere/bee/blob/master/pkg/p2p/libp2p/libp2p_test.go#L80, especially to avoid the need for defers.